warehouse/packaging/views.py

```diff
@@ -53,6 +57,53 @@ def project_detail(project, request):
     return release_detail(release, request)
 
 
+@view_config(
+    route_name="packaging.project_latest",
+    context=Project,
+)
+def project_latest(project, request):
+    return HTTPTemporaryRedirect(
+        request.route_path(
+            "packaging.release",
+            name=project.name,
+            version=project.latest_version.version,
+        )
+    )
+
+
+@view_config(
+    route_name="packaging.project_latest_stable",
+    context=Project,
+)
+def project_latest_stable(project, request):
+    release = project.latest_stable_version
+
+    if release:
+        return HTTPTemporaryRedirect(
+            request.route_path(
+                "packaging.release",
+                name=project.name,
+                version=project.latest_stable_version.version,
+            )
+        )
+    else:
+        return HTTPNotFound()
+
+
+@view_config(
+    route_name="packaging.project_latest_unstable",
+    context=Project,
+)
+def project_latest_unstable(project, request):
+    return HTTPTemporaryRedirect(
+        request.route_path(
+            "packaging.release",
+            name=project.name,
+            version=project.latest_unstable_version.version,
```

Comment:
what if `project.latest_unstable_version` is None ?

Reply:
Fixed in dc838308